cmd/contour/serve.go

```diff
@@ -205,6 +208,14 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 	informers = registerEventHandler(informers, dynamicInformers.ForResource(projectcontour.TLSCertificateDelegationGVR).Informer(), eh)
 	informers = registerEventHandler(informers, coreInformers.Core().V1().Services().Informer(), eh)
 
+	if ctx.UseServiceAPITypes {
+		log.Info("Enabling EXPERIMETNAL SERVICE API TYPES")
+		informers = registerEventHandler(informers, dynamicInformers.ForResource(serviceapis.GroupVersion.WithResource("gatewayclasses")).Informer(), eh)
+		informers = registerEventHandler(informers, dynamicInformers.ForResource(serviceapis.GroupVersion.WithResource("gateways")).Informer(), eh)
+		informers = registerEventHandler(informers, dynamicInformers.ForResource(serviceapis.GroupVersion.WithResource("httproutes")).Informer(), eh)
+		informers = registerEventHandler(informers, dynamicInformers.ForResource(serviceapis.GroupVersion.WithResource("tcproutes")).Informer(), eh)
+	}
```

Comment:
This will be more readable if you flip the arguments to `registerEventHandler` to take varargs of informers:
```
func registerEventHandler(informers []cache.SharedIndexInformer, eh cache.ResourceEventHandler, informersToAdd ...cache.SharedIndexInformer) []cache.SharedIndexInformer {
...
}
```

Reply:
Okay, that's done.